components/omega/doc/design/VertCoord.md

```diff
@@ -0,0 +1,264 @@
+The $z$ location of the interfaces is needed to compute the geopotential gradient and the sea level.
+The variable will be registered with the `IOStreams` framework to enable it to be output during runtime.
+
+### 2.3 Requirement: Compute and store the vertical level bounds for each horizontal cell
+
+All tendency terms require a variable vertical loop range to account for variation in bottom elevation and ice shelf cavities.
+These bounds will be computed on construction in the vertical coordinate module and used for setting the loop bounds for active levels in calculations over the vertical.
+
+### 2.4 Requirement: Compute and store the geopotential
+
+The geopotential will be computed and stored within the vertical coordinate module.
+Initially this will be calculated by summing together the $z$ location times $g$.
+In the future, tidal potential and self attraction and loading will also have optional (default off) contributions to the geopotential.
+
+
+### 2.5 Requirement: Compute and store the desired thickness used in the $p^\star$ vertical coordinate
+
+The $p^\star$ vertical coordinate specifies a desired layer thickness that stretches the initial thicknesses by the bottom pressure anomaly.
+This involves adding a vertically-weighted bottom pressure perturbation to the reference (initial) thickness.
+This desired thickness will be computed and stored within the vertical coordinate module.
+The vertical advection algorithm will use this desired thickness to compute the vertical transport.
+
+### 2.6 Desired: General Arbitrary Lagrangian Eulerian (ALE) coordinate support
+
+In future versions of Omega, the $p^\star$ coordinate will be extended to a more general ALE coordinate
+
+### 2.7 Desired: Vertical Lagrangian remapping (VLR) support
+
+The vertical coordinate module should retain flexibility to support VLR in future versions of Omega.
+
+## 3 Algorithmic Formulation
+In the layered non-Boussinesq equations solved in Omega (see [V0 governing equation document](OmegaV1GoverningEqns) for details), the prognostic variable for cell $i$ and layer $k$ is the pseudo thickness, $\tilde{h}_{i,k}$, so that the geometric thickness (in meters) is a diagnostic variable defined as:
+
+$$ \Delta z_{i,k} = \rho_0 \alpha_{i,k} \tilde{h}_{i,k}. $$
+
+The pressure at vertical cell interfaces is the found by summing the pseudo thicknesses:
+
+$$  p_{i,k+1/2} = p_{i}^{surf} + g\rho_0 \sum_{k^\prime=1}^k \tilde{h}_{i,k^\prime}. $$
+
+and at the midpoint by
+
+$$ p_{i,k} = p_{i}^{surf} + g\rho_0 \sum_{k'=1}^{k-1} \tilde{h}_{i,k'} + \frac{1}{2} g\rho_0 \tilde{h}_{i,k} $$
+
+The $z$ location of cell interfaces is found by summing the pseudo thicknesses from the bottom multiplied by the specific volume:
+
+$$ z_{i,k}^{top} = z_i^{floor} + \rho_0 \sum_{k^\prime=k}^{K_{max}} \alpha_{i,k^\prime} \tilde{h}_{i,k^\prime}, $$
```

Comment:
Shouldn't this be $z_{i,k+1/2}$, similar to $p_{i,k+1/2}$ above?

Reply:
yes, thanks!